mesa/model.py

```diff
@@ -51,12 +52,22 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self.running = True
         self.schedule = None
         self.current_id = 0
-        self.agents: defaultdict[type, dict] = defaultdict(dict)
+        self._agents: defaultdict[type, dict] = defaultdict(dict)
+
+    @property
+    def agents(self) -> AgentSet:
+        all_agents = set()
```

Comment:
I know the discussion about a set have been all over the place, but I thought we create AgentSet specifically to be deterministic. Using a set here destroys this advantage, no? I think we can use some itertools to chain the individual types of the _agents dict and pass that to AgentSet, so we don't need to create an intermediate list

Reply:
I agree that this should not be a set. I have focussed on AgentSet so far, but I will take a look at this next.